src/flows/components/panel/FlowPanel.tsx

```diff
@@ -140,17 +142,27 @@ const FlowPanel: FC<Props> = ({id, children, controls}) => {
     updatePanelFocus(false)
   }
 
+  const showResults = ['inputs', 'test', 'transform'].includes(
+    PIPE_DEFINITIONS[flow.data.get(id).type].family
+  )
+
   if (
     flow.readOnly &&
     !/^(visualization|markdown)$/.test(flow.data.get(id).type)
   ) {
     return null
   }
+
   return (
     <ClickOutside onClickOutside={handleClickOutside}>
       <div className={panelClassName} onClick={handleClick} ref={panelRef}>
         <FlowPanelHeader id={id} controls={controls} />
         <div className="flow-panel--body">{children}</div>
+        {showResults && (
```

Comment:
It feels like this could exist in `Results` and make the component itself responsible for rendering?

Reply:
JK that might make it confusing since the `Results` component is responsible for some pagination visibility